modules/internal/ChapelDomain.chpl

```diff
@@ -1283,6 +1283,271 @@ module ChapelDomain {
       return _newArray(x);
     }
 
+    /*
+      This manager is returned by ``unsafeAssign()`` and can be used in
+      managed blocks to help resize arrays of non-default-initializable
+      elements.
+    */
+    record unsafeAssignManager {
+      pragma "no doc"
+      var _lhsInstance;
+
+      pragma "no doc"
+      var _lhsPid: int;
+
+      pragma "no doc"
+      var _rhsInstance;
+
+      pragma "no doc"
+      var _rhsPid: int;
+
+      pragma "no doc"
+      var _oldLhsDomainCopy: domain;
+
+      pragma "no doc"
+      param _checks: bool;
+
+      pragma "no doc"
+      var _isActiveManager: bool;
+
+      pragma "no doc"
+      iter _lhsBaseArrays() {
+        for arr in _lhsInstance._arrs do yield arr;
```

Comment:
I commented in my overall feedback, but I think here, we should check whether `arr` is a view, and yield only if it is not. I don't think we need to do anything for views here. (I am still trying to understand some of the related behavior with array views, so I am about 90 percent sure about this).

You might want to rename the iterator to make it more self-documenting. ie, `arraysToCheck` or something. I don't like that either too much.

Reply:
Changed the name to `_arraysOverLhsDom`